### PR TITLE
[imt][NFC] Improve docs for RSlotStack

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

File: core/imt/inc/ROOT/RSlotStack.hxx [hunk 1]
```diff
@@ -18,10 +18,15 @@
 namespace ROOT {
 namespace Internal {
 
-/// This is an helper class to allow to pick a slot resorting to a map
-/// indexed by thread ids.
-/// WARNING: this class does not work as a regular stack. The size is
-/// fixed at construction time and no blocking is foreseen.
+/// A thread-safe stack of N indexes (0 to size - 1).
```
Comment (Axel-Naumann) [inline core/imt/inc/ROOT/RSlotStack.hxx:21]:
```suggestion
/// A thread-safe, constant-size stack of N indexes (0 to size - 1).
```

File: core/imt/inc/ROOT/RSlotStack.hxx [hunk 2]
```diff
@@ -18,10 +18,15 @@
...
+/// RSlotStack can be used to safely assign a "processing slot" number to
+/// each thread in multi-thread applications.
+/// In release builds, pop and push operations are unchecked, potentially
+/// resulting in undefined behavior if more slot numbers than avaialble are
+/// requested.
```
Comment (Axel-Naumann) [inline core/imt/inc/ROOT/RSlotStack.hxx:26]:
```suggestion
/// accessed.
```

File: core/imt/inc/ROOT/RSlotStack.hxx [hunk 3]
```diff
@@ -18,10 +18,15 @@
...
 
-/// This is an helper class to allow to pick a slot resorting to a map
-/// indexed by thread ids.
-/// WARNING: this class does not work as a regular stack. The size is
-/// fixed at construction time and no blocking is foreseen.
+/// A thread-safe stack of N indexes (0 to size - 1).
+/// RSlotStack can be used to safely assign a "processing slot" number to
+/// each thread in multi-thread applications.
+/// In release builds, pop and push operations are unchecked, potentially
+/// resulting in undefined behavior if more slot numbers than avaialble are
```
Comment (Axel-Naumann) [inline core/imt/inc/ROOT/RSlotStack.hxx:25]:
```suggestion
/// resulting in undefined behavior if slot numbers outside the size are
```

**Review by Axel-Naumann — APPROVED** [review]
See comments

File: core/imt/inc/ROOT/RSlotStack.hxx (hunk 1 above)
Comment (eguiraud) [reply core/imt/inc/ROOT/RSlotStack.hxx:21]: the stack is not constant size. we pop slot indexes from it and then put them back. after construction, there are `size` indexes available and it's UB to call `GetSlot` more than `size` times without calling `ReturnSlot` in the meanwhile between.

File: core/imt/inc/ROOT/RSlotStack.hxx (hunk 2 above)
Comment (eguiraud) [reply core/imt/inc/ROOT/RSlotStack.hxx:26]: slot numbers are not accessed  you can request a new slot number from the `RSlotStack` with `GetSlot` and that pops a number out of a `std::stack<unsigned int>`.

**Review by vepadulano — APPROVED** [review]
LGTM thanks!

File: core/imt/inc/ROOT/RSlotStack.hxx (hunk 3 above)
Comment (Axel-Naumann) [reply core/imt/inc/ROOT/RSlotStack.hxx:25]: That leaves us with:
```suggestion
/// resulting in undefined behavior if more slot numbers than available are
```

**Pull request merged**